api/config/v1/imex.go

```diff
@@ -0,0 +1,98 @@
+/**
+# Copyright 2024 NVIDIA CORPORATION
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+**/
+
+package v1
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"math"
+)
+
+const (
+	ImexChannelEnvvar = "NVIDIA_IMEX_CHANNELS"
+
+	DefaultChannelStrategyAuto     = DefaultChannelStrategy("auto")
+	DefaultChannelStrategyEnabled  = DefaultChannelStrategy("enabled")
+	DefaultChannelStrategyDisabled = DefaultChannelStrategy("disabled")
+
+	defaultChannelStrategyInvalid = DefaultChannelStrategy("invalid")
```

Comment:
Do we really need all of these options?

If we accept my suggestion above, the existence of a channel in the config implies "auto" and the lack of one implies "disabled" using the terminology defined here. I don't really see the need for a strict "enabled", but that could be added later via a single boolean option of `imex.must-inject` or something.

Reply:
The behaviour is as follows:
* `disabled`: no IMEX-specific modifications are made.
* `enabled`: IMEX-modifications are made regardless of whether the device plugin has access to the device nodes from within the container. This means that workload containers will fail if they are started and the device nodes are not present. (the exact implementation of this in libnvidia-container would have to be checked, but at least in CDI the OCI spec cannot be constructed since we can't query the relevant information from the host device node).
* `auto`: The device plugin checks for the presence of the (default) channel and behaves as `enabled` if it can be found, if not we run without `imex` support. This allows for the GPU Operator to set a single value to `auto` and expect the device plugin to be sane wherever it is installed, meaning that it makes the decision as to whether IMEX is enabled or not.

I don't really see the benefit of having two possibilities `auto`, `disabled` for one option and a second option (`must-inject`) which only has meaning in conjunction with `auto`.

I'm happy to talk through scenarios that we need to cover here to see whether I'm missing something though.

I think we **can** remove the `invalid` value which is internal-only.